app/models/case/sar/offender.rb

```diff
@@ -1,33 +1,72 @@
-class Case::SAR::Offender < Case::SAR::Standard
+class Case::SAR::Offender < Case::Base
   class << self
     def type_abbreviation
       'OFFENDER'
     end
   end
 
-  validates :prison_number, presence: true
-  validates_presence_of :name, :third_party_relationship, if: -> { third_party }
+  def self.searchable_fields_and_ranks
+    super.merge(
+        {
+            subject_full_name:     'B'
+        }
+    )
+  end
+
   jsonb_accessor :properties,
-                  prison_number: :string,
-                  subject_aliases: :string,
-                  previous_case_numbers: :string,
-                  other_subject_ids: :string,
-                  date_of_birth_dd: :string,
-                  date_of_birth_mm: :string,
-                  date_of_birth_yyyy: :string,
-                  subject_type: :string,
-                  received_date_dd: :string,
-                  received_date_mm: :string,
-                  received_date_yyyy: :string,
-                  reply_method: :string
+                 prison_number: :string,
+                 subject_full_name: :string,
+                 escalation_deadline: :date,
+                 internal_deadline: :date,
+                 subject_aliases: :string,
+                 previous_case_numbers: :string,
+                 other_subject_ids: :string,
+                 external_deadline: :date,
+                 date_of_birth: :date,
+                 third_party: :boolean,
+                 third_party_relationship: :string,
+                 subject_type: :string,
+                 received_date: :date,
+                 reply_method: :string,
+                 flag_for_disclosure_specialists: :boolean
```

Comment:
Can the list of attributes be ordered alphabetically and do any of them require default values?

Reply:
Changed, and no, I don't think so...?